src/cryptoadvance/specter/wallet.py

```diff
@@ -189,6 +189,45 @@ def fetch_transactions(self):
                     tx["blockheight"] = current_blockheight - tx["confirmations"] + 1
         ##################### Remove until here after dropping Core v0.19 support #####################
         self._transactions.add(txs)
+        if self.use_descriptors:
+            while (
+                len(
+                    [
+                        tx
+                        for tx in self._transactions
+                        if self._transactions[tx]["category"] != "send"
+                        and not self._transactions[tx]["address"]
+                    ]
+                )
+                != 0
```

Comment:
In #1316 i'm currently migrating to v0.21.1 and the tests are hanging in this endless loop.
How does this loop is suppose to end, @Sjors ?

Reply:
@stepansnigirev a Python strangle?